File: backend/src/models/ReceiptGeometry.ts
interface BoundingBox {
  Width: number;
  Height: number;
  Left: number;
  Top: number;
}

interface Point {
  X: number;
  Y: number;
}

export interface ReceiptGeometry {
  PK: string; // RECEIPT#receipt_id
  SK: string; // FIELD#FIELDNAME_TYPE (e.g., FIELD#SUBTOTAL_VALUE)
  receipt_id: string;
  field_name: string; // e.g., "subtotal", "tax", "total"
  field_type: 'label' | 'value';
  text: string;
  confidence: number;
  bounding_box: BoundingBox;
  polygon: Point[];
  created_at: string;
}

File: api/CostcoReceipts.Api/Data/Entities/ReceiptGeometry.cs
namespace CostcoReceipts.Api.Data.Entities;

public class ReceiptGeometry
{
    public long Id { get; set; }
    public string ReceiptId { get; set; } = string.Empty;
    public string FieldName { get; set; } = string.Empty;
    public string FieldType { get; set; } = string.Empty;
    public string Text { get; set; } = string.Empty;
    public double Confidence { get; set; }

    public double BoundingBoxWidth { get; set; }
    public double BoundingBoxHeight { get; set; }
    public double BoundingBoxLeft { get; set; }
    public double BoundingBoxTop { get; set; }

    public string PolygonJson { get; set; } = "[]";

    public DateTime CreatedAt { get; set; }

    public Receipt Receipt { get; set; } = null!;
}
